<?xml version="1.0"?>
<launch>

  <arg name = "object" default = "box" />
  <arg name = "simulation" default = "false" />
  <arg name = "use_joy" default = "true" />

  <!-- ad-hoc method for quad type -->
  <arg name="delta_angle" default="0.02"/>
  <arg name="contact_torque_thresh" default="1.8" />
  <arg name="extra_module_comepsention_delay" default="0.05"/>
  <arg name="measure_object_mass" default="true"/>
  <arg name="estimate_mass_du" default="4.0"/>
  <arg name="joint1_init_angle" default="0.8"/>
  <arg name="joint3_init_angle" default="0.8"/>
  <arg name="joint2_relax_rate" default="8.0"/> <!-- this value is measured from rosbag of realmachine, 15~20, thus we choose smaller value 8-10, 8 is better -->


  <node pkg="vectoring_thrust_grasp" type="grasping_motion_node" name="grasping_motion_node" output="screen" ns="dragon">
    <rosparam file="$(find vectoring_thrust_grasp)/config/object/$(arg object).yaml" command="load" />

    <param name="delta_angle" value="$(arg delta_angle)"/>
    <param name="contact_torque_thresh" value="$(arg contact_torque_thresh)" />
    <param name="extra_module_comepsention_delay" value="$(arg extra_module_comepsention_delay)"/>
    <param name="measure_object_mass" value="$(arg measure_object_mass)"/>
    <param name="estimate_mass_du" value="$(arg estimate_mass_du)"/>
    <param name="joint1_init_angle" value="$(arg joint1_init_angle)"/>
    <param name="joint3_init_angle" value="$(arg joint3_init_angle)"/>
    <param name="joint2_relax_rate" value="0" if="$(arg simulation)"/>
    <param name="joint2_relax_rate" value="$(arg joint2_relax_rate)" unless="$(arg simulation)"/>
  </node>

  <node pkg="vectoring_thrust_grasp" type="spawn_target_object.py" name="spawn_target_object" output="screen"  if="$(arg simulation)" ns="dragon">
    <rosparam file="$(find vectoring_thrust_grasp)/config/object/$(arg object).yaml" command="load" />
    <param name="sdf_file_path" value="$(find vectoring_thrust_grasp)/config/model_template.sdf"/>
    <param name="static_object" value="false"/>
    <param name="joint1_init_angle" value="$(arg joint1_init_angle)"/>
    <param name="joint3_init_angle" value="$(arg joint3_init_angle)"/>
  </node>

  <node pkg="vectoring_thrust_grasp" type="joystick_operation.py" name="joystick_operation" output="screen" if="$(arg use_joy)" />

</launch>
